src/test/java/com/iexec/common/chain/eip712/EIP712ChallengeTests.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2022 IEXEC BLOCKCHAIN TECH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.common.chain.eip712;
+
+import com.iexec.common.utils.HashUtils;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.Test;
+import org.web3j.crypto.Hash;
+import org.web3j.utils.Numeric;
+
+import java.math.BigInteger;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Slf4j
+public class EIP712ChallengeTests {
+
+    @Test
+    void shouldReturnJsonString() {
+        Challenge challenge = Challenge.builder().challenge("challenge").build();
+        EIP712Domain domain = new EIP712Domain("COMMON", "1", 15L, null);
+        EIP712Challenge eip712Challenge = new EIP712Challenge(domain, challenge);
+        assertThat(eip712Challenge.toJsonString())
+                .isNotEmpty()
+                .isEqualTo("{\"types\":{\"EIP712Domain\":[" +
+                        "{\"name\":\"name\",\"type\":\"string\"}," +
+                        "{\"name\":\"version\",\"type\":\"string\"}," +
+                        "{\"name\":\"chainId\",\"type\":\"uint256\"}" +
+                        "]," +
+                        "\"Challenge\":[{\"name\":\"challenge\",\"type\":\"string\"}]}," +
+                        "\"domain\":{\"name\":\"COMMON\",\"version\":\"1\",\"chainId\":15}," +
+                        "\"message\":{\"challenge\":\"challenge\"}," +
+                        "\"primaryType\":\"Challenge\"}");
+        String domainType = "EIP712Domain(string name,string version,uint256 chainId)";
+        String domainSeparator = HashUtils.concatenateAndHash(
```

Comment:
Not sure whether it should be dynamically computed or static. Usually, when writing unit tests, I think we should not rely on production code as it may broke at some point.

Reply:
OK, I see what you mean. I had not read the last line of the test, where you check those values against a static value. So it's alright, sorry :)

Reply:
No problem, resolving this thread